### PR TITLE
Request body sanity fix

### PR DESCRIPTION
**Pull request by kunagpal**
cc @abhijitkane 

Closes #442

### CODE REVIEW

File: lib/collection/request.js [hunk 1]
```diff
@@ -83,6 +83,11 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
         _.has(options, 'method') && (this.method = _.isString(options.method) ? options.method.toUpperCase() :
             (options.method || DEFAULT_REQ_METHOD));
 
+        if (_.has(options, 'body')) {
+            _.isString(options.body) && (options.body = { mode: 'raw', raw: options.body });
```
Comment (kamalaknn) [inline lib/collection/request.js]: Not sure about the self-correction here.

File: lib/collection/request.js [hunk 2]
```diff
@@ -83,6 +83,11 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
...
+            !_.isPlainObject(options.body) && (delete options.body);
```
Comment (kamalaknn) [inline lib/collection/request.js]: Why are we mutating options here?

File: lib/collection/request.js (hunk 1 above)
Comment (kunagpal) [reply lib/collection/request.js]: As discussed offline.

File: lib/collection/request.js [hunk 3]
```diff
@@ -83,17 +83,19 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
         _.has(options, 'method') && (this.method = _.isString(options.method) ? options.method.toUpperCase() :
             (options.method || DEFAULT_REQ_METHOD));
 
+        var body;
```
Comment (kamalaknn) [inline lib/collection/request.js]: Should be moved to top of the function.

File: lib/collection/request.js [hunk 4]
```diff
@@ -83,17 +83,19 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
...
+
         if (_.has(options, 'body')) {
-            _.isString(options.body) && (options.body = { mode: 'raw', raw: options.body });
-            !_.isPlainObject(options.body) && (delete options.body);
+            body = options.body;
```
Comment (kamalaknn) [inline lib/collection/request.js]: `body = _.isString(body) ? { mode: 'raw', raw: body } : options.body;`

Comment (kunagpal) [reply lib/collection/request.js]: Before: 240K ops/s
After: 220K ops/s

File: test/unit/request-body.test.js [hunk 5]
```diff
@@ -79,6 +79,18 @@ describe('RequestBody', function () {
...
     });
 
+    it('should work with string bodies correctly', function () {
+        var body = new RequestBody('foo');
+        expect(body).to.have.property('mode', 'raw');
+        expect(body).to.have.property('raw', 'foo');
+    });
+
+    it('should reject invalid body types', function () {
+        expect(new RequestBody([])).to.not.have.property('body');
```
Comment (kamalaknn) [inline test/unit/request-body.test.js]: This will always pass 🤔

Comment (kunagpal) [reply test/unit/request-body.test.js]: Yes. The collection SDK didn't validate for non-standard body options before.

Comment (kamalaknn) [reply test/unit/request-body.test.js]: why are we asserting `new RequestBody().body`? this will always be empty right? even for valid body types?

Comment (kunagpal) [reply test/unit/request-body.test.js]: Incorrect test; has been fixed.

**Pull request merged**